cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/WorkflowScheduleUtils.java

```diff
@@ -0,0 +1,133 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+
+package io.cdap.cdap.datapipeline.service;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.schedule.ProgramStatusTriggerInfo;
+import io.cdap.cdap.etl.api.lineage.field.FieldOperation;
+import io.cdap.cdap.etl.common.BasicArguments;
+import io.cdap.cdap.etl.common.FieldOperationTypeAdapter;
+import io.cdap.cdap.etl.proto.v2.ArgumentMapping;
+import io.cdap.cdap.etl.proto.v2.PluginPropertyMapping;
+import io.cdap.cdap.etl.proto.v2.TriggeringPipelineId;
+import io.cdap.cdap.etl.proto.v2.TriggeringPropertyMapping;
+import io.cdap.cdap.internal.io.SchemaTypeAdapter;
+import org.slf4j.Logger;
+
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Map;
+
+public final class WorkflowScheduleUtils {
+  private static final Gson GSON = new GsonBuilder()
+    .registerTypeAdapter(Schema.class, new SchemaTypeAdapter())
+    .registerTypeAdapter(FieldOperation.class, new FieldOperationTypeAdapter()).create();
+  private static final String RESOLVED_PLUGIN_PROPERTIES_MAP = "resolved.plugin.properties.map";
+  private static final Type STAGE_PROPERTIES_MAP = new TypeToken<Map<String, Map<String, String>>>() { }.getType();
+
+  private WorkflowScheduleUtils() {
+  }
+
+  /**
+   * Parse the arguments mapping from the pipeline schedule.
+   *
+   * @param triggerInfo {@link ProgramStatusTriggerInfo} with plugin properties
+   * @param propertiesMapping  {@link TriggeringPropertyMapping} for using the plugin
+   * @param logger  {@link Logger} the logger
+   * @param compositeTriggerEnabled  whether the composite trigger is enabled
+   */
+  public static void processSchedulePropertiesMapping(Map<String, String> mappings,
+                                                      ProgramStatusTriggerInfo triggerInfo,
+                                                      TriggeringPropertyMapping propertiesMapping,
+                                                      Logger logger,
+                                                      boolean compositeTriggerEnabled) {
+    TriggeringPipelineId pipelineId = new TriggeringPipelineId(
+      triggerInfo.getNamespace(),
+      triggerInfo.getApplicationName());
+
+    BasicArguments triggeringArguments = new BasicArguments(
+      triggerInfo.getWorkflowToken(),
+      triggerInfo.getRuntimeArguments());
+
+    // Get the value of every triggering pipeline arguments specified in the propertiesMapping and update newRuntimeArgs
+    List<ArgumentMapping> argumentMappings = propertiesMapping.getArguments();
```

Comment:
is this just copy-pasted code? It's a bit hard to follow, so if there are any differences from before can you comment on them?

Reply:
yes, this is partially copied code, but also with some new code checking on the pipeline ID